core/src/test/resources/json/shaped_recipe_list.json

```diff
@@ -0,0 +1,62 @@
+[ {
+  "key" : {
+    "namespace" : "minecraft",
+    "key" : "test"
+  },
+  "result" : {
+    "material" : "STONE",
+    "amount" : 32
+  },
+  "shape" : [ "aaa", " b ", " b " ],
+  "group" : "",
+  "ingredients" : {
+    " " : null,
+    "a" : {
+      "@clazz" : "MaterialChoiceWrapper",
+      "choices" : [ "STONE", "COBBLED_DEEPSLATE" ]
+    },
+    "b" : {
+      "@clazz" : "MaterialChoiceWrapper",
+      "choices" : [ "STICK" ]
+    }
+  }
+}, {
+  "key" : {
+    "namespace" : "minecraft",
+    "key" : "test2"
+  },
+  "result" : {
+    "material" : "BONE",
+    "amount" : 1
+  },
+  "shape" : [ " a ", " a ", " a " ],
+  "group" : "",
+  "ingredients" : {
+    " " : null,
+    "a" : {
+      "@clazz" : "ExactChoiceWrapper",
+      "choices" : [ {
+        "material" : "DIAMOND",
+        "amount" : 2
+      } ]
+    }
+  }
+}, {
+  "key" : {
+    "namespace" : "minecraft",
+    "key" : "test3"
+  },
+  "result" : {
+    "material" : "CHEST",
+    "amount" : 9
+  },
+  "shape" : [ " a ", " a ", " a " ],
+  "group" : "",
+  "ingredients" : {
+    " " : null,
```

Comment:
See above, probably good to add one shaped recipe without the Strings.EMPTY -> null ingredient to test how that gets handled.